examples/xoxo/xoxo.py

```diff
@@ -0,0 +1,183 @@
+                display_board[row][col] = Cell.x
+                turns += 1
+            elif cell == Cell.o.name:
+                solver.add_assertion(Equals(board[row][col], Int(Cell.o.value)))
+                display_board[row][col] = Cell.o
+
+def get_win_assertion(player):
+    return [
+        # rows
+        Equals(Plus(board[0]), Int(player * sq_size)), 
+        Equals(Plus(board[1]), Int(player * sq_size)), 
+        Equals(Plus(board[2]), Int(player * sq_size)),
+        # cols
+        Equals(Plus([row[0] for row in board]), Int(player * sq_size)),
+        Equals(Plus([row[1] for row in board]), Int(player * sq_size)),
+        Equals(Plus([row[2] for row in board]), Int(player * sq_size)),
+        # diags
+        Equals(Plus([board[0][0], board[1][1], board[2][2]]), Int(player * sq_size)),
+        Equals(Plus([board[0][2], board[1][1], board[2][0]]), Int(player * sq_size))
+        ]
+
+# assertions for players to win
+o_win_assertions = get_win_assertion(Cell.o.value)
+x_win_assertions = get_win_assertion(Cell.x.value)
+
+logger.debug(solver.assertions)
+
+def print_board():
+    for row in display_board:
+        logger.info(" ".join([cell.name.replace('s','.') for cell in row]))
+
+def print_solver_board():
+    logger.debug("solver board")
+    for row in board:
+        logger.debug(" ".join([Cell(int(solver.get_py_value(cell))).name.replace('s','.') for cell in row]))
+
+def get_a_move(player):
+    logger.debug("looking for a move for %s" % player)
+    # prioritise center position
+    if display_board[1][1] == Cell.s and Cell(int(solver.get_py_value(board[1][1]))) == player:
+        return(1,1)
+    for r, row in enumerate(board):
+        for c, cell in enumerate(row):
+            if display_board[r][c] == Cell.s and Cell(int(solver.get_py_value(cell))) == player:
+                return(r,c)
+
+def convert_num_to_indices(num):
+    row = int(num/sq_size)
+    col = num % sq_size
+    return(row,col)
+
+def already_played(row, col):
+    if display_board[row][col] == Cell.s:
+        return False
+    return True
+
+def play_move(player, row, col):
+    display_board[row][col] = player
+    solver.add_assertion(Equals(board[row][col], Int(player.value)))
+
+while True:
+
+    # get user input and handle errors
+    logger.info("-" * 40)
+    logger.info("turn %d" % turns)
+    print_board()
+    try:
+        next_cell = int(raw_input("type a cell (1-9):")) - 1
+    except ValueError:
+        continue
+    if next_cell < 0 or next_cell > 8:
+        continue
+
+    # convert index to rows, cols, check if space is free
+    row, col = convert_num_to_indices(next_cell)
+    if(not already_played(row, col)):
+        play_move(Cell.x, row, col)
+    else:
+        logger.info("that cell is already taken")
+        continue
+
+    turns += 1
+
+    # assertions for next move
+    x_next_move_assertion = Equals(Int((turns+1)*Cell.x.value + (turns*Cell.o.value)), Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))
```

Comment:
Wrap this into a function to make it more clear.

Reply:
as in:
x_next_move_assertion = get_x_next_move_assertion(turns)
?
Not sure that makes it any clearer.

Reply:
ok, I think this can be broken down to make more clear, but it is ok like this if there is no better way.
`` Plus(Plus(board[0]), Plus(board[1]), Plus(board[2])))`` this is basically the sum over the whole board, right? Maybe this can be made more clear?

``Equals(Int((turns+1)*Cell.x.value + (turns*Cell.o.value))`` what is the intuition behind this.